File: 박혜은/AC.py
import sys

input = sys.stdin.readline

# .reverse() -> 시간 오래 걸림
# 따라서 토글 방식으로 reverse여부를 판단하고, 마지막에 한 번만 뒤집기
def solve(command, n, p):
    rev = False
    if n == 0 and 'D' in command:
        return "error"
    for char in command:
        if char == 'R':
            rev = not rev
        elif char == 'D':
            if len(p) == 0:
                return "error"
            if rev:
                p.pop()
            else:
                p.pop(0)
    if rev:
        p.reverse()  # 변수에 할당하면 안된다(None 반환)
    # 이것때메 틀린게 아니라고 해줘..
    str = '[]'
    if p:
        str = '[' + ','.join(p) + ']'
    return str


T = int(input().rstrip())
for tc in range(1, T + 1):
    command = list(input().rstrip())
    n = int(input().rstrip())
    data = input().rstrip().strip('[]')  # 1,2,3
    arr = list(data.split(',')) if data else []  # ['1','2','3']
    result = solve(command, n, arr)
    print(result)

File: 박혜은/algo-java/src/AC.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class AC {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int T = Integer.parseInt(br.readLine());

        for (int tc = 1; tc <= T; tc++) {
            String cmd = br.readLine();
            List<Character> command = new ArrayList<>();
            for (char ch : cmd.toCharArray()) {
                command.add(ch);
            }
            int n = Integer.parseInt(br.readLine());
            String line = br.readLine().replaceAll("\\[|\\]", "");
            List<String> arr = new ArrayList<>();
            if (!line.isEmpty()) {
                // 배열을 리스트로 바꾸는 이유는 remove나 reverse 등을 써야하기 때문
                String[] parts = line.split(",");
                Collections.addAll(arr, parts);
            }
            String result = solve(command, n, arr);
            System.out.println(result);
        }
    }

    static String solve(List<Character> command, int n, List<String> arr) {
        boolean rev = false;
        if (n == 0 && command.contains('D')) return "error";
        for (char ch : command) {
            if (ch == 'R') {
                rev = !rev;
            } else if (ch == 'D') {
                if (arr.isEmpty()) return "error";
                if (rev) arr.remove(arr.size() - 1);
                else arr.remove(0);
            }
        }
        if (rev) {
            Collections.reverse(arr);
        }
        String str = "[]";
        if (!arr.isEmpty()) {
            StringBuilder sb = new StringBuilder();
            sb.append("[");
            for (int i = 0; i < arr.size(); i++) {
                sb.append(arr.get(i));
                if (i != arr.size() - 1) {
                    sb.append(",");
                }
            }
            sb.append("]");
            str = sb.toString();
        }
        return str;
    }
}
